[Parsers] HDBProcedure name is not correctly extracted

Having a comment containing the **Procedure** keyword, before the procedure SQL, results into an exception, due to wrong name extraction:

**test.hdbprocedure:**
```sql
-- File Description: Procedure to update all Table1.
-- Dependencies : none
-- Created by: John Doe
-- Updated at: 24.12.2021
-- History:
-- 20.03.2018 - John - Initial Version
--  
--
PROCEDURE "DBADMIN"."test::Table1" ( )
    ...
```

**Exception:**
```java
Caused by: org.h2.jdbc.JdbcSQLDataException: Value too long for column "DS_NAME CHARACTER VARYING(255)": "U&'to update all Table1. ... (316)"; SQL statement:
INSERT INTO "XSK_DATA_STRUCTURES" ("DS_LOCATION", "DS_NAME", "DS_TYPE", "DS_HASH", "DS_CREATED_BY", "DS_CREATED_AT") VALUES (?, ?, ?, ?, ?, ?) [22001-206]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:525) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:496) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.message.DbException.get(DbException.java:227) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.message.DbException.getValueTooLongException(DbException.java:341) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.value.Value.getValueTooLongException(Value.java:2572) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.value.Value.convertToVarchar(Value.java:1261) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.value.Value.convertTo(Value.java:1117) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.value.Value.convertForAssignTo(Value.java:1092) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.table.Column.validateConvertUpdateSequence(Column.java:369) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.table.Table.convertInsertRow(Table.java:917) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.command.dml.Insert.insertRows(Insert.java:167) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.command.dml.Insert.update(Insert.java:135) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.command.dml.DataChangeStatement.update(DataChangeStatement.java:61) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.command.CommandContainer.update(CommandContainer.java:173) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.command.Command.executeUpdate(Command.java:252) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209) ~[h2-2.0.206.jar:2.0.206]
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:169) ~[h2-2.0.206.jar:2.0.206]
	at org.apache.commons.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:97) ~[commons-dbcp2-2.2.0.jar:2.2.0]
	at org.apache.commons.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:97) ~[commons-dbcp2-2.2.0.jar:2.2.0]
	at org.eclipse.dirigible.database.persistence.processors.entity.PersistenceInsertProcessor.insert(PersistenceInsertProcessor.java:102) ~[dirigible-database-persistence-6.1.20.jar:na]
	... 23 common frames omitted
```

**Exception reason:**
- At [XSKDataStructuresCoreService.java#L52](https://github.com/SAP/xsk/blob/main/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/service/XSKDataStructuresCoreService.java#L52) the **name** parameter is containing the comment and not the correct procedure name

--- a/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/parser/hdbprocedure/XSKHDBProcedureParser.java
+++ b/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/parser/hdbprocedure/XSKHDBProcedureParser.java
@@ -28,7 +28,11 @@ public class XSKHDBProcedureParser implements XSKDataStructureParser {
   @Override
   public XSKDataStructureHDBProcedureModel parse(XSKDataStructureParametersModel parametersModel) throws XSKDataStructuresException {
     XSKDataStructureHDBProcedureModel hdbProcedure = new XSKDataStructureHDBProcedureModel();
-    hdbProcedure.setName(XSKHDBUtils.extractProcedureNameFromContent(parametersModel.getContent(), parametersModel.getLocation()));
+
+    parametersModel.setContent(parametersModel.getContent().replaceAll("--.*", "").trim()); // remove comment lines
+
+    hdbProcedure.setName(XSKHDBUtils.extractProcedureNameFromContent(parametersModel.getContent(),
+        parametersModel.getLocation()));
     hdbProcedure.setLocation(parametersModel.getLocation());
     hdbProcedure.setType(getType());
     hdbProcedure.setHash(DigestUtils.md5Hex(parametersModel.getContent()));

